lib/cli-engine.js

```diff
@@ -459,6 +474,30 @@ class CLIEngine {
              */
             this._lintResultCache = new LintResultCache(cacheFile, this.config);
         }
+
+        // setup special filter for fixes
+        if (this.options.fixTypes && this.options.fixTypes.length > 0) {
```

Comment:
I might be misunderstanding this logic, but won't this still apply fixes if `originalFix` is `false`? The logic in this `if` statement checks if `originalFix` is a function, but if `originalFix` is not a function it seems to always apply fixes.

It might be better to do something like this:

```suggestion
        if (this.options.fix && this.options.fixTypes && this.options.fixTypes.length > 0) {
```

Reply:
Oh wow, nice catch. `cli` errors if you use `--fix-type` without `--fix`, so that's probably why I missed it, but it should definitely be checked here, too. I'll update and add a test.